src/controllers/legacycontrollersettings.h

```diff
@@ -259,11 +259,16 @@ class LegacyControllerNumberSetting
     /// and a strictly positive step, strictly less than max..
     /// @return true if valid
     bool valid() const override {
+        static constexpr bool typeSize = sizeof(long) == sizeof(int);
         return AbstractLegacyControllerSetting::valid() &&
                 m_defaultValue >= m_minValue && m_savedValue >= m_minValue &&
                 m_editedValue >= m_minValue && m_defaultValue <= m_maxValue &&
                 m_savedValue <= m_maxValue && m_editedValue <= m_maxValue &&
-                m_stepValue > 0 && m_stepValue < m_maxValue;
+                m_stepValue > 0 &&
+                (typeSize ||
```

Comment:
To be confirmed, but I read that some platform use the same byte size for `int` and `long` and since the default min and max uses the int limits, without a larger type, this would lead to an overflow.

Reply:
In case of an int a step value can't be bigger than the half of the range. If we assume 8 bit, we may have 
-128 ... 127 and the maximum Step that can be expressed is 127. 

So you may check: 
m_minValue <= 0 && m_minValue + m_stepValue <= m_maxValue &&
m_maxValue >= 0 && m_maxValue - m_stepValue >= sm_minValue

Reply:
And m_maxValue - m_stepValue < m_maxValue for the unsigned case ...

Reply:
and put it in a utility function please. The long condition chain is already not super easy to follow.

Reply:
> And m_maxValue - m_stepValue < m_maxValue for the unsigned case ...

Not sure which usecase you are referring to?